api/util/registerUser.js

```diff
@@ -0,0 +1,127 @@
+const User = require('../models/User');
+const config = require('../config/config');
+
+/**
+ * Register a new user.
+ * @param {Object} newUser - The user that is to be registered
+ * @param {(Object} userToAdd - The information about the user that is
+ *                              registering
+ */
+async function registerUser(userToAdd){
+  let res = {
+    userSaved: true,
+    message: '',
+    status: 'OK'
+  };
+  if (userToAdd.email && userToAdd.password) {
```

Comment:
This is good to have, but there is an issue with the code. You initialize `res.userSaved` to be true by default, but what if there is no email or password? Right now, the function would just skip the `if` block and return `res`.

Either initialize `userSaved` to `false` or have an `else` block which sets it to false